retrymanager_default.go

```diff
@@ -40,9 +41,9 @@ func (rc *retryControllerDefault) isRetriableError(err error) bool {
 		errors.Is(err, ErrCollectionManifestOutdated)
 }
 
-func (rc *retryControllerDefault) ShouldRetry(err error) (time.Duration, bool) {
+func (rc *retryControllerDefault) ShouldRetry(ctx context.Context, err error) (time.Duration, bool, error) {
```

Comment:
Why do we need the added return error here if it's always nil?

Reply:
It's nil in this case, but it's not nil in the other retry controller for consistency.